src/transformers/modeling_utils.py

```diff
@@ -1275,6 +1275,13 @@ def tie_weights(self):
             if hasattr(module, "_tie_weights"):
                 module._tie_weights()
 
+    def load_adaptive_weights(self):
+        """
+        Load adaptive weights after state dict has been loaded. If required this method should be overridden by derived
+        class.
+        """
+        pass
```

Comment:
I'd raise an exception here if it's called, otherwise it fails silently
```suggestion
        raise NotImplementedError
```

Reply:
Yeah not 100% sure about the design here, but it seems much more in line with:
- https://github.com/huggingface/transformers/blob/c5454eba9eac00a3e7d0a46a3d25aacd43187f1e/src/transformers/modeling_utils.py#L1257
and 
- https://github.com/huggingface/transformers/blob/c5454eba9eac00a3e7d0a46a3d25aacd43187f1e/src/transformers/modeling_utils.py#L1242

Think something like `if hasattr(self, "load_adaptive_weights")` is also not great. 

Also it is a bit questionable whether `load_adaptive_weights` is general enough to warrant to be in `modeling_utils.py`, but there is no other way really for the functionality `from_pretrained(..., target_lang="...")`

@LysandreJik @sgugger wdyt?

Reply:
For now I would override the `_tie_encoder_decoder_weights` method in Wav2Vec2 only and not add those changes in modeling_utils. If things change and we get lots of models with adapaters, we can revisit the decision and do something like this, but I'd wait for it to be necessary.

Reply:
That's pretty hacky, but yeah ok for me. Will add a big comment that we do this to not introduce a new API